azurelinuxagent/common/utils/archive.py

```diff
@@ -38,7 +38,7 @@
 
 ARCHIVE_DIRECTORY_NAME = 'history'
 
-_MAX_ARCHIVED_STATES = 100
+_MAX_ARCHIVED_STATES = 50
```

Comment:
Previously I increased this value from 50 to 100 because I was saving waagent_status.json to a separate directory. Since now I am saving it to the same directory as the goal state, I am reverting the value to the original 50